### PR TITLE
Create Screenshots for 17.1

### PR DESCRIPTION
**Pull request by jkmassel**
Fixes screenshots as part of the `17.1 `release

To test:
1. Checkout this PR
2. Run `rake mocks`
3. In Xcode, run the `WordPressScreenshotGeneration` test on a blank simulator.

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

**PR submission checklist:**

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -358,9 +358,9 @@ end
 ## ===================
 ##
 def wordpress_mocks
-  pod 'WordPressMocks', '~> 0.0.9'
+  # pod 'WordPressMocks', '~> 0.0.9'
   # pod 'WordPressMocks', :git => 'https://github.com/wordpress-mobile/WordPressMocks.git', :commit => ''
-  # pod 'WordPressMocks', :git => 'https://github.com/wordpress-mobile/WordPressMocks.git', :branch => ''
+  pod 'WordPressMocks', :git => 'https://github.com/wordpress-mobile/WordPressMocks.git', :branch => 'fix/reader'
```
Comment (mokagio) [inline Podfile]: I assume before merging this PR we should have a new version of the mocks library out, yeah?

File: fastlane/ScreenshotFastfile [hunk 2]
```diff
@@ -24,21 +24,17 @@ platform :ios do
...
     scan(
-      workspace: File.join(fastlane_directory, "../../WordPress.xcworkspace"),
+      workspace: WORKSPACE_PATH,
       scheme: "WordPressScreenshotGeneration",
       build_for_testing: true,
       derived_data_path: derived_data_path,
     )
 
-    languages = "da de-DE en-AU en-CA en-GB es-ES fr-FR id it ja ko no nl-NL pt-BR pt-PT ru sv th tr zh-Hans zh-Hant en-US".split(" ")
+    languages = "ru".split(" ")
```
Comment (mokagio) [inline fastlane/ScreenshotFastfile]: I believe this is debug code that sneaked into the PR?

File: fastlane/Snapfile [hunk 3]
```diff
@@ -1,59 +0,0 @@
...
-# end
-
-# teardown_language do |lang, device|
-#   puts "Finished with #{lang} on #{device}"
-# end
-
-# teardown_device do |device|
-#   puts "Cleaning device #{device}"
-#   system("./cleanup.sh")
-# end
```
Comment (mokagio) [inline fastlane/Snapfile:59]: +1 for moving the settings in this file to the dedicated `Fastfile` for screenshots. We don't need to spread information between files, in particular in when using a tool that does _a lot_ behind the scenes like Fastlane. Thanks.

**Review by mokagio — CHANGES_REQUESTED** [review]
The tests passed from Xcode:

image

They also worked fine (as far as I can see) via Fastlane (with the hardcoded `"ru"` locale only still) once I [updated the helper](https://github.com/wordpress-mobile/WordPress-iOS/commit/9e4ad9f68b680fcb5c53a0d93ce755fbf3ec696b) as prompted by Fastlane. Feel free to cherry-pick.

screencapture-file-Users-gio-Developer-a8c-wpios-fastlane-screenshots-screenshots-html-2021-05-11-16_15_44

Requesting changes because of the hardcoded `"ru"` locale.

File: fastlane/ScreenshotFastfile (hunk 2 above)
Comment (jkmassel) [reply fastlane/ScreenshotFastfile]: Ah yep, thanks!!

File: Podfile (hunk 1 above)
Comment (jkmassel) [reply Podfile]: Yes indeed – this is updated!

File: fastlane/ScreenshotFastfile [hunk 4]
```diff
@@ -34,7 +34,7 @@ platform :ios do
       derived_data_path: derived_data_path,
     )
 
-    languages = "ru".split(" ")
+    languages = "da de-DE en-AU en-CA en-GB es-ES fr-FR id it ja ko no nl-NL pt-BR pt-PT ru sv th tr zh-Hans zh-Hant en-US".split(" ")
```
Comment (mokagio) [inline fastlane/ScreenshotFastfile]: Random question: have you considered making this an array? Would it help maintaining the setup? 🤔

**Review by mokagio — APPROVED** [review]
I did not run the screenshots automation locally with the new changes, but by looking at the commits I'm 99.9% confident the result would be the same as my previous run.

I didn't see the [snapshot helper update commit](https://github.com/wordpress-mobile/WordPress-iOS/commit/9e4ad9f68b680fcb5c53a0d93ce755fbf3ec696b). I don't think that's a blocker, but I wonder if we should add it? On my machine, I wasn't able to run the `screenshots` lane before applying those changes 🤔

File: fastlane/ScreenshotFastfile (hunk 4 above)
Comment (jkmassel) [reply fastlane/ScreenshotFastfile]: Yeah, good question – ideally we'd pull this from a global "list of locales we support" that's aligned between screenshots, metadata, glotpress, etc. For now I'm inclined to leave it, but I look forwarding to having it be a lot nicer, as you say!

**Pull request merged**